### PR TITLE
[Hot Fix] Ensure process_dataloader is called when tpu_cores > 1 to use Parallel DataLoader

### PR DESCRIPTION
**Pull request by tchaton**
## What does this PR do?

There was a regression in speed when using the TPU accelerator. After inspection, we realised that during the accelerator refactor we missed migrating the `process_dataloader` hook to the `TrainingTypePlugin`. This meant that we never wrapped the model when TPU cores > 1 in a parallel data-loader. This PR adds the correct logic.

A bigger discussion needs to take place in finding these corner cases which would be shown with regression/benchmarking tests. In the interim we should fix this particular error for release.

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by awaelchli — APPROVED** [review]

**Review by Borda — APPROVED** [review]
can we get a test for it?

**Review by SeanNaren — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -269,6 +269,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Fixed passing wrong strings for scheduler interval doesn't throw an error ([#5923](https://github.com/PyTorchLightning/pytorch-lightning/pull/5923))
 
 
+- Fixed missing `process_dataloder` call for `TPUSpawn` ([#6015](https://github.com/PyTorchLightning/pytorch-lightning/pull/6015))
```
Comment (SeanNaren) [inline CHANGELOG.md:272]:
```suggestion
- Fixed missing `process_dataloader` call for `TPUSpawn` when in distributed mode ([#6015](https://github.com/PyTorchLightning/pytorch-lightning/pull/6015))
```

**Comment by SeanNaren** [comment]
I made an edit to the main post, I think we need to have some benchmark tests for TPUs that ensure that we stay within reasonable limits for speed/accuracy. We test for model change, but would be good to have a general test for catching these regressions

**Review by williamFalcon — APPROVED** [review]

**Comment by williamFalcon** [comment]
nice... yeah we need a test for this so we don't regress in the future
